Scripts/generate-test-config.sh

```diff
@@ -1,14 +1,269 @@
 #!/bin/bash
 
-DESTINATION=$1
+# vim: set ts=2 sw=2 sts=2 et:
 
-if [[ -z $DESTINATION ]] ; then
-  echo "Destination not specified" >&2
+##################################################
+# SCRIPT METADATA
+
+SCRIPT_NAME=$(basename $0)
+SUMMARY="Generates a testconfiguration.json file for use in Amplify and AWS mobile SDK integration tests"
+SYNOPSIS="${SCRIPT_NAME} <android|ios> [options]"
+
+
+##################################################
+# DEFAULTS & DECLARATIONS
+##################################################
+LOG_LEVEL_INFO=0
+LOG_LEVEL_DEBUG=1
+LOG_LEVEL_TRACE=2
+LOG_LEVEL=0
+
+
+##################################################
+# FUNCTIONS
+##################################################
+
+# Logging. Note that all logging output goes to STDERR
+
+function log {
+  msg="$(date) $*"
+  echo $msg >&2
+}
+
+function log_debug {
+  if [[ $LOG_LEVEL -lt LOG_LEVEL_DEBUG ]] ; then
+    return
+  fi
+
+  log "[D]:: $*"
+}
+
+# Always logged
+function log_error {
```

Comment:
To the same benefit of modularization in other programming languages, you could extract your logging stuff to a `logging.sh`, and then source it into this script with `source logging.sh` or `. logging.sh`.

Reply:
I considered but rejected that, in favor of keeping the script and its functional dependencies self-contained. (I also don't know of a good strategy for dependency management for bash--we'll usually be invoking these scripts from a directory other than the one the script itself is contained in, which makes discovery of the "library" files tricky. I guess I could do something like:

```bash
readonly SCRIPT_ROOT=$(dirname $0)
source ${SCRIPT_ROOT}/logging.sh
```

For now, I'm going to postpone this refactor until we get another bash script that needs these functions, at which point I'll extract the library (or move to a different language :) )